splink/comparison.py

```diff
@@ -67,10 +68,16 @@ def __init__(self, comparison_dict, settings_obj: Settings = None):
         for cl in comparison_level_list:
             if isinstance(cl, ComparisonLevel):
                 cl.comparison = self
-            elif settings_obj is None:
-                cl = ComparisonLevel(cl, self)
             else:
-                cl = ComparisonLevel(cl, self, sql_dialect=settings_obj._sql_dialect)
+                # cl is a dict
+                # TODO: remove support for this
```

Comment:
For this I just mean at the level of `ComparisonLevel`. Fine to keep support for users using a dict instead of a `CustomLevel` if they want, but I think we should convert `dict` -> `CustomLevel` when we get the opportunity, so that `ComparisonLevel` (which is not what the user should be interacting with) takes only specific parameters.

i.e. `ComparisonLevelCreator` can be a dict instead, but `ComparisonLevel` can't